lib/source_opam_config.ml

```diff
@@ -224,9 +270,12 @@ let set_field set var opam_file =
 let set ~opam_monorepo_cwd { global_vars; repositories; opam_provided }
     opam_file =
   let opam_monorepo_cwd = opam_monorepo_cwd_from_root opam_monorepo_cwd in
+  let repositories =
+    Opam_repositories_url_rewriter.rewrite_out ~opam_monorepo_cwd repositories
```

Comment:
There's no reason to forget as it's handled in a single place and the parsing code is not otherwise exposed.

Also worth noting that if you ever forget it, the entire test suite will crash as it all relies on this!